source/1_engine/rex_engine/src/entrypoint.cpp

```diff
@@ -1,12 +1,51 @@
 #include "rex_engine/entrypoint.h"
 
+#include "rex_engine/cmdline.h"
+#include "rex_engine/diagnostics/logging/logger_config.h"
+#include "rex_engine/filesystem/vfs.h"
+#include "rex_std/array.h"
+#include "rex_std/chrono.h"
+#include "rex_std/internal/exception/exit.h"
 #include "rex_std/string.h"
+#include "rex_std/thread.h"
 
 namespace rex
 {
   namespace internal
   {
-    void pre_app_entry(REX_MAYBE_UNUSED const char8* cmdLine) {}
+    void pre_app_entry(REX_MAYBE_UNUSED const char8* cmdLine)
+    {
+      cmdline::init(rsl::string_view(cmdLine));
+
+      // if a user wants to know the arguments for the executable, we want to perform as minimal setup as possible.
+      // we just initialize the commandline, print what's possible and exit the program
+      if(cmdline::get_argument("help"))
+      {
+        cmdline::print_args();
+        rsl::exit(0);
+      }
+
+      // if we want to debug executables without a debugger, we need to break early
+      // so we can attach a debugger and continue from then on
+      // we'll have a timer in place to break for 2 minutes, if no debugger is attached
+      // we close down the program
+      if (cmdline::get_argument("BreakOnBoot"))
+      {
+        using namespace rsl::chrono_literals;
+        auto i = 1s;
+        while (i < 10min)
+        {
+          rsl::this_thread::sleep_for(1s);
+          ++i;
```

Comment:
I would encapsulate the timing functionality, as this might become useful elsewhere

Reply:
Where else do you see this exact code useful?
If you want to wait, you often sleep for x seconds
I can't do that here as it's possible a debugger gets attached

Reply:
For instance, AI might not update every frame. Objects in the distance might run at 12 fps rather than 60 so there could be some update controller which manages this that waits for an update

Reply:
yes, that's fine, but that shouldn't follow the above code, they wait for x seconds in 1 go, they don't iterate on the amount of time they're waiting.

Reply:
also I'd think such a system has a dependency on the engine being initialized and this update controller being up and running etc.
here I want to have a simple timer, allow someone to attach a debugger and immediately continue executing without having to wait for the remaining time to finish.
if no debugger gets attached in 10min, I want the app to close or it'd keep running forever.
it's important almost nothing else gets done before it reaches this point or it defeats the purpose of it

Reply:
Okay, true and we cannot create a callback ( a promise ) that is called when x amount of time is passed so this implementation can become a bit more generic?

Reply:
we can, but it'd have to look the same as above which I don't think we're gonna need.
this is basically "spinlock", it checks something, if not true yet, sleeps and repeats this n times.
if we want to generalise this, that's fine, but that's outside of the scope of this PR I'd say

Reply:
We'll discuss this later it's fine for now ;-)